templates/index.html

```diff
@@ -145,5 +163,53 @@ <h3>Videos</h3>
   });
 
   </script>
-
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.0.4/socket.io.js"></script>
+  <script type='module'>
+    import { Spinner } from '/static/js/lib/spin.js'
+    
+    /* This block is required to add a spinner if the user refreshes and the job is not done */
+    {% for job_dict in session['jobs'] %}
+      {% set job = q.fetch_job(job_dict['id']) %}
+      {% if job and job.meta.get('status') and job.meta.get('status') != 'Done' %}
+        new Spinner().spin(document.getElementById('{{ job_dict["id"] }}'));
+      {% endif %}
+    {% endfor %}
+
+    var socket = io.connect('http://ec2-3-237-25-198.compute-1.amazonaws.com:5000/');
```

Comment:
Please add this as an environment variable, read it in the python file, and pass it to the template.

Reply:
https://github.com/MLDSAI/VisionTracing/pull/2/commits/c9e4704ac7fc4c9e22947d03db8dbc368affa643